### PR TITLE
Feat/navbar by roles fe

### PR DESCRIPTION
**Pull request by oyakovytskyi**
## Describe your changes

- Navigation items are now based on user roles:
Superadmin: Displays company-related items.
Admin/Dispatcher: Displays orders and route management items.
- Route Protection: Update route protection logic
- New Orders: Show the number of unassigned orders at navbar 

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://codecraftersintership.atlassian.net/browse/SCRUM-123)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

photo_2024-12-05_07-45-23

### CODE REVIEW

**Review by AlexeyBezrebriy — APPROVED** [review]

File: src/components/Layout/RootLayout/index.tsx [hunk 1]
```diff
@@ -1,44 +1,61 @@
...
-      {
-        segment: 'company-list',
-        title: t('navigation.companyList'),
-        icon: <ListIcon />,
-      },
-    ],
-    [t]
-  );
+  const NAVIGATION: Navigation = React.useMemo(() => {
+    if (role === 'superadmin') {
```
Comment (lyudad) [inline src/components/Layout/RootLayout/index.tsx]: 1.1.5
roles should be stored in enums in the general constants

File: src/components/Layout/RootLayout/index.tsx [hunk 2]
```diff
@@ -1,44 +1,61 @@
...
+      return [
+        {
+          segment: 'company-list',
+          title: t('navigation.companyList'),
+          icon: <ListIcon />,
+        },
+      ];
+    } else if (role === 'admin' || role === 'dispatcher') {
```
Comment (lyudad) [inline src/components/Layout/RootLayout/index.tsx]: 1.1.5

File: src/components/Layout/RootLayout/index.tsx (hunk 1 above)
Comment (oyakovytskyi) [reply src/components/Layout/RootLayout/index.tsx]: done

File: src/components/Layout/RootLayout/index.tsx (hunk 2 above)
Comment (oyakovytskyi) [reply src/components/Layout/RootLayout/index.tsx]: changed

**Pull request merged**